pkg/collector/providers/file.go

```diff
@@ -123,13 +102,54 @@ func (c *FileConfigProvider) String() string {
 	return "File Configuration Provider"
 }
 
-func collectDir(parentPath string, folder os.FileInfo) []check.Config {
+func (c *FileConfigProvider) collectEntry(entry os.FileInfo, path string, checkName string) (check.Config, bool, error) {
```

Comment:
I had a hard time understanding `collectEntry` applied to files. Could you add documentation to collectEntry and collectDir ?